### ISSUE TITLE
Normalized TSPLs not in use

### ISSUE THREAD

**Issue by TrevorWinstral** [Jordylek/VolatilityIsMostlyPathDependent#1]
Hi, looking over the code I found that although functions for the normalized TSPLs are implemented, they are not used. In your paper, it is claimed that this normalization is necessary for having a unique set of linear regression coefficients. This is a simple fix:
1. In the `empirical_study_TSPL` file, the function `compute_features_TSPL` should use `normalized_TSPL` instead of  `shifted_power_law`;
2.  In the `jacobian` functions, `deriv_alpha_shift_power_law` should instead give the parital derivative of the normalized TSPL rather than the non-normalized TSPL. The same goes for `deriv_delta_shift_power_law`. These partial derivatives are:
```math
\frac{\partial}{\partial \alpha}: \quad TSPL_{\alpha, \delta}(t) \left[ \delta \log( (t + \alpha) \delta) + \frac{1}{\alpha -1} \right] 
```
```math
\frac{\partial}{\partial \delta}: \quad TSPL_{\alpha, \delta}(t) \left[ \frac{2 \delta \alpha + t (\alpha - 1)}{(t+\delta)\delta} \right]
```
This change, in my implementation with 
`(    vol=vix,
    index=spx,
    p=1,
    tspl=True,
    setting=[(1,1), (2,1/2)],
    train_start_date = pd.to_datetime("2000-01-03"),
    test_start_date = pd.to_datetime("2015-02-23"),
    test_end_date = pd.to_datetime("2017-04-12"),
    max_delta=400)`
I get an $R^2$ of 0.866, and when swapping VIX for Oxford-Man RV, an $R^2$ of 0.322 (each on the test data), this compares to the previous values of 0.855 and 0.654.

**Comment by Jordylek** [comment]
Hi, thanks a lot for your comments. Indeed, one could use the `normalized_TSPL` when doing the optimization. We opted for using the non-normalized one and normalize then kernels at the end. They should both lead to the same results. With the hyperparameters you gave, I obtain the exact same scores and optimal parameters when using the normalized and unnormalized kernels. I have a train $R^2$ of 0.943 and a test $R^2$ of 0.768 on the VIX. Note that in the paper, we have a different train and test set, so we should not expect the same $R^2$.

Besides, I find different answers when computing the partial derivatives.

$$
\frac{\partial}{\partial \alpha} : TSPL_{\alpha, \delta}(t) \left[ \frac{1}{\alpha-1} - \log(t/\delta +1)\right]
$$

 $$
\frac{\partial}{\partial \delta}: \quad TSPL_{\alpha, \delta}(t) \left[ \frac{\alpha - 1}{\delta} - \frac{\alpha}{t+\delta} \right] =
\quad TSPL_{\alpha, \delta}(t) \left[ \frac{t\alpha - (t+\delta)}{\delta(t+\delta)} \right] 
$$

Maybe that is what lead to the discrepancy?

**Comment by TrevorWinstral** [comment]
I see that using non-normalized yields the same fits (I just found the same in my testing). I also agree that my partial derivatives are wrong.

Now I have changed this in my code (and used `max_delta=1000` as you do) and I get test $R^2$'s of 0.855 on the VIX (same dates as in your ipynb) and 0.331 on the Oxford-Man RV (my archive only goes up to April 2017) using a 0.875:0.125 train:test split. 

So you were correct that the partials were the problem on the VIX data, but I am interested how you got 0.654 $R^2$ on the Oxford-Man data.

**Comment by Jordylek** [comment]
That is a very good point. What are your train and test RMSE? The 0.654 $R^2$ is for our test data (2019-2022). Changing the test set will undoubtedly change the score. You can refer to figure B.2 (in appendix). We plotted the RMSE for different test set, you can see that for the RV of SPX, we have a test RMSE of about 0.05 for 2015-2017. That would be a better point for comparison
